stdlib/public/Concurrency/TaskStatus.cpp

```diff
@@ -795,7 +802,7 @@ void AsyncTask::pushInitialTaskName(const char* _taskName) {
   // TODO: Copy the string maybe into the same allocation at an offset or retain the swift string?
   char* taskNameCopy = reinterpret_cast<char*>(
       _swift_task_alloc_specific(this, strlen(_taskName) + 1/*null terminator*/));
-  (void) strcpy(/*dst=*/taskNameCopy, /*src=*/_taskName);
+  (void) _strcpy(/*dst=*/taskNameCopy, /*src=*/_taskName);
```

Comment:
Shouldn't this continue to use `strcpy` when not in  `SWIFT_STDLIB_TASK_TO_THREAD_MODEL_CONCURRENCY`?

Reply:
I was a bit worried if it's the right condition but yeah, let's give it a shot.

Updated to:

```
#if SWIFT_STDLIB_TASK_TO_THREAD_MODEL_CONCURRENCY
// Embedded runtime does not seem to have strcpy available so we provide our own.
char *__strcpy(char *dest, const char *src) {
  return reinterpret_cast<char *>(::memcpy(dest, src, ::strlen(src) + 1));
}
auto _strcpy = __strcpy;
#else
auto _strcpy = strcpy;
#endif
```

Reply:
I'd have been more comfortable with a #define, rather than something with static storage like this

Reply:
Ok, I'll do a define then

Reply:
Seems we could use `strncpy` will do that instead